waku/persistence/store.go

```diff
@@ -300,17 +300,20 @@ func (d *DBStore) Validate(env *protocol.Envelope) error {
 
 // Put inserts a WakuMessage into the DB
 func (d *DBStore) Put(env *protocol.Envelope) error {
-	stmt, err := d.db.Prepare("INSERT INTO message (id, receiverTimestamp, senderTimestamp, contentTopic, pubsubTopic, payload, version) VALUES ($1, $2, $3, $4, $5, $6, $7)")
+
+	stmt, err := d.db.Prepare("INSERT INTO message (id, messageHash, storedAt, timestamp, contentTopic, pubsubTopic, payload, version) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)")
 	if err != nil {
 		d.metrics.RecordError(insertFailure)
 		return err
 	}
 
-	cursor := env.Index()
-	dbKey := NewDBKey(uint64(cursor.SenderTime), uint64(cursor.ReceiverTime), env.PubsubTopic(), env.Index().Digest)
+	storedAt := env.Message().GetTimestamp()
+	if storedAt == 0 {
```

Comment:
Why can't storedAt always be receiverTimestamp? 
Since msgTimestamp is anyways stored separately, wouldn't it be better to use this field for time at which message is received.

Also, wondering if this field is really required since store query should ideally only rely on senderTimestamp for better query response gaurantees across different store nodes. Maybe i am missing something. Why was the receiverTime included in store query at all since different nodes in the network can receive message at differnt times (considering granularity of 1 sec).

Reply:
The message timestamp is an optional value so we can't rely on it for queries. I agree that the way it's done is confusing, in part because the `storedAt` column shouldnt be called that, because the data it contains does not refer just to the receiver timestamp, but also to the sender timestamp as seen in nwaku here:  https://github.com/waku-org/nwaku/blob/ebad0385ef61c55cf0699a114fb64cb6e88d9680/waku/waku_store/client.nim#L159-L160 in which the storedAt will contain the sender timestamp if available, otherwise the current timestamp.

Perhaps this is something that we could rename to something different? like `queryTimestamp` or dunno. (cc: @Ivansete-status @jm-clius)

Regarding the second point, `storedAt` is currently being used for retrieiving query results within a time range. See https://github.com/waku-org/nwaku/blob/ebad0385ef61c55cf0699a114fb64cb6e88d9680/waku/waku_archive/driver/sqlite_driver/queries.nim#L266-L274